### PR TITLE
[cxx-interop] Apply typedef -> enum patch to method param types as well.

### PR DESCRIPTION
**Pull request by zoecarver**
Basically just applying https://github.com/apple/swift/pull/42431 to ObjC method param types.

### CODE REVIEW

**Comment by zoecarver** [comment]
@swift-ci please test.

**Comment by zoecarver** [comment]
@swift-ci Please Test Source Compatibility

File: lib/ClangImporter/ImportType.cpp [hunk 1]
```diff
@@ -2734,10 +2734,28 @@ ImportedType ClangImporter::Implementation::importMethodParamsAndReturnType(
...
-      importType(resultType, resultKind, addImportDiag,
-                 allowNSUIntegerAsIntInResult, Bridgeability::Full,
-                 getImportTypeAttrs(clangDecl), OptionalityOfReturn);
+
+
+  ImportedType importedType;
+  if (auto typedefType = dyn_cast<clang::TypedefType>(resultType.getTypePtr())) {
+    if (isUnavailableInSwift(typedefType->getDecl())) {
+      if (auto clangEnum = findAnonymousEnumForTypedef(SwiftContext, typedefType)) {
+        // If this fails, it means that we need a stronger predicate for
```
Comment (plotfi) [inline lib/ClangImporter/ImportType.cpp:2743]: Could this comment be an assert? Should it?

Comment (plotfi) [reply lib/ClangImporter/ImportType.cpp:2743]: Ah sorry, nvm. just saw the assert.

File: lib/ClangImporter/ImportType.cpp [hunk 2]
```diff
@@ -2847,7 +2865,21 @@ ImportedType ClangImporter::Implementation::importMethodParamsAndReturnType(
     ImportTypeKind importKind = ImportTypeKind::Parameter;
     ImportDiagnosticAdder paramAddDiag(*this, clangDecl, param->getLocation());
     Type swiftParamTy;
-    bool paramIsIUO;
+    bool paramIsIUO = false;
+    if (auto typedefType = dyn_cast<clang::TypedefType>(paramTy.getTypePtr())) {
```
Comment (plotfi) [inline lib/ClangImporter/ImportType.cpp:2869]: Can this and the previous addition be merged into one function or lambda?

File: test/Interop/Cxx/enum/anonymous-with-swift-name-objc-module-interface.swift [hunk 3]
```diff
@@ -0,0 +1,18 @@
+// RUN: %target-swift-ide-test -print-module -module-to-print=AnonymousWithSwiftName -I %S/Inputs -source-filename=x -enable-experimental-cxx-interop | %FileCheck %s
+//
+// REQUIRES: objc_interop
+
+// CHECK: class ColorMaker {
+// CHECK:   class func makeColor(withOptions opts: SOColorMask)
+// CHECK:   func makeColor(withOptions opts: SOColorMask)
```
Comment (plotfi) [inline test/Interop/Cxx/enum/anonymous-with-swift-name-objc-module-interface.swift:7]: I suspect you will have to update these now that https://github.com/apple/swift/pull/42412 is merged.

**Comment by zoecarver** [comment]
Windows test failures are unrelated. Merging.

File: lib/ClangImporter/ImportType.cpp (hunk 2 above)
Comment (zoecarver) [reply lib/ClangImporter/ImportType.cpp:2869]: Good idea. I will do it in a follow-up nfc.

**Pull request merged**